jdaviz/core/template_mixin.py

```diff
@@ -68,3 +76,219 @@ def _mxn_update_plugin_opened(self, new_value):
         app_state = self.app.state
         tray_names_open = [app_state.tray_items[i]['name'] for i in app_state.tray_items_open]
         self.plugin_opened = app_state.drawer and self._registry_name in tray_names_open
+
+
+class BasePluginComponent(HubListener):
+    """
+    This Base class handles attaching traitlets from the plugin itself to logic
+    handled within the component, support for caching and clearing caches on properties,
+    and common properties for accessing the app, etc.
+    """
+    def __init__(self, plugin, **kwargs):
+        self._plugin_traitlets = {k: v for k, v in kwargs.items() if v is not None}
+        self._plugin = plugin
+        super().__init__()
+
+    def __getattr__(self, attr):
+        if attr[0] == '_' or attr not in self._plugin_traitlets.keys():
+            return super().__getattribute__(attr)
+
+        return getattr(self._plugin, self._plugin_traitlets.get(attr))
+
+    def __setattr__(self, attr, value, force_super=False):
+        if attr[0] == '_' or force_super or attr not in self._plugin_traitlets.keys():
+            return super().__setattr__(attr, value)
+
+        return setattr(self._plugin, self._plugin_traitlets.get(attr), value)
+
+    def _clear_cache(self, *attrs):
+        """
+        provide convenience function to clearing the cache for cached_properties
+        """
+        for attr in attrs:
+            if attr in self.__dict__:
+                del self.__dict__[attr]
+
+    def add_observe(self, traitlet_name, handler):
+        self._plugin.observe(handler, traitlet_name)
+
+    @property
+    def app(self):
+        return self._plugin.app
+
+    @property
+    def hub(self):
+        return self._plugin.hub
+
+    @cached_property
+    def spectrum_viewer(self):
+        return self._plugin.app.get_viewer("spectrum-viewer")
+
+
+class SpectralSubsetSelect(BasePluginComponent):
+    """
+    Traitlets (in the object, custom traitlets in the plugin):
```

Comment:
Bulleted list needs a blank line here to render properly.
```suggestion
    Traitlets (in the object, custom traitlets in the plugin):

```